### ISSUE TITLE
`cd` to Git repo root

### ISSUE THREAD

**Issue by 9999years** [Canop/broot#760]
Hello! I'd like to be able to quickly change my directory to the Git repository root, e.g. with a `gr` verb.

I could define a verb with an `internal = ":focus {git-root}"`, but there's no way for me to get the `{git-root}` value. It's easy enough to call `git rev-parse --show-toplevel`, but I don't believe the `internal` verb attribute supports shell execution.

Should we implement this specific feature (e.g. a `{git-root}` substitution variable) or a more general one (e.g. `$(...)` shell interpolation in `internal` or `external` verb definition attributes)?

**Comment by Canop** [comment]
This is an interesting question but I don't think there's a need to change broot for this.

I defined this function in my `.bashrc`:

```
function go_to_git_root() {
    cd $1
    cd $(git rev-parse --show-toplevel)
}
```

Then I added this verb definition in my `verbs.hjson`:

```
    {
        invocation: gr
        external: ["go_to_git_root", "{directory}"]
        from_shell: true
    }
```

It seems to do what you want.

**Comment by 9999years** [comment]
Oh, but I don't want to leave broot, I want to `:focus` the repo root for further searching.

**Comment by Canop** [comment]
Oh, sorry, I had misunderstood you.

**Comment by partiallyordered** [comment]
Woo! I can't believe I came here to make this very issue (after having it in my to-do list for quite some time) and here it is right at the top of the list.

While I'm here, thank you @Canop, the more I use Broot, the better it gets! (And of course, the better you make it, the better it gets!).

I recognise this is a verbose +1, hope it's not too annoying but you are very welcome to delete.

**Comment by 9999years** [comment]
I made an attempt at this in #761.

**Comment by 9999years** [comment]
Closed with #761 merged.
